flang/lib/Semantics/check-declarations.cpp

```diff
@@ -2623,6 +2628,43 @@ void CheckHelper::CheckGlobalName(const Symbol &symbol) {
   }
 }
 
+void CheckHelper::CheckProcedureAssemblyName(const Symbol &symbol) {
+  if (!IsProcedure(symbol) || symbol != symbol.GetUltimate())
+    return;
+  const std::string *bindName{symbol.GetBindName()};
+  const bool hasExplicitBindingLabel{
+      symbol.GetIsExplicitBindName() && bindName};
+  if (hasExplicitBindingLabel || IsExternal(symbol)) {
```

Comment:
Please ensure that this logic won't elicit an inappropriate error for a clash between `BIND(C,NAME="")` and `SUBROUTINE _`.

Reply:
I added a test showing this is OK and a few more cases that should not elicit errors.